jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -1163,4 +1163,180 @@ else if (l.size() == 1)
+        return regrCountLong(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrCountLong()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Long>> regrCountLong(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            long result = 0;
+            for (T t : l)
+                result += t==null ? 0:1;
+            return Optional.of(result);
+        });
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgXDouble()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Double>> regrAvgXDouble() {
+        return regrAvgXDouble(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgXDouble()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrAvgXDouble(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            else if (l.size() == 1)
+                return Optional.of(0.0);
+            return Optional.of(avg0(l,functionX));
+        });
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgYDouble()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Double>> regrAvgYDouble() {
+        return regrAvgYDouble(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrAvgYDouble()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrAvgYDouble(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(toList(), l -> {
+            if (l.isEmpty())
+                return Optional.empty();
+            else if (l.size() == 1)
+                return Optional.of(0.0);
+            return Optional.of(avg0(l,functionY));
+        });
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrSxxDouble()</code> function.
+     */
+    public static  Collector<Tuple2<Double,Double>, ?, Optional<Double>> regrSxxDouble() {
+        return regrSxxDouble(t -> t.v1, t -> t.v2);
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the <code>regrSxxDouble()</code> function.
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrSxxDouble(ToDoubleFunction<? super T> functionX, ToDoubleFunction<? super T> functionY) {
+        return collectingAndThen(
+                collectors(covarianceDouble(functionX, functionX),
+                        count()),
+                t -> !t.v1.isPresent() || t.v2==0 ? Optional.empty() : Optional.of(t.v1.get() * t.v2 )
+        );
+    }
```

Comment:
At first, I thought that looks like `REGR_SXY()`, not `REGR_SXX()`, because of the `COVAR_POP()` usage, but then I realised, you passed `functionX` twice, which makes this effectively `varianceDouble()` (which was implemented recently by your colleague). Better use that for clarity.

Reply:
But again (and here we go, getting PRs for ill-specified issues is what we get when we ill-specify issues): Is this really useful? The `REGR_SXX()`, `REGR_SXY()`, and `REGR_SYY()` functions are some weird auxiliary functions in SQL. This is just multiplying the count with variance of covariance.

But both variance and covariance are some formula divided by the count, so this cancels out. It would then be more efficient to have a collector that calculates `REGR_SXX()`, `REGR_SXY()`, and `REGR_SYY()`, and base the other functions on this.